src/table.ts

```diff
@@ -23,44 +23,35 @@ import * as through from 'through2';
 import {Operation as GaxOperation} from 'google-gax';
 import {Json} from './codec';
 import {Database} from './database';
-import {SpannerClient as s} from './v1';
 import {PartialResultStream, Row} from './partial-result-stream';
-import {ReadRequest, TimestampBounds} from './transaction';
+import {
+  ReadRequest,
+  TimestampBounds,
+  CommitResponse,
+  ReadResponse,
+  ReadCallback,
+  Transaction,
+  CommitCallback,
+} from './transaction';
 import {google as databaseAdmin} from '../proto/spanner_database_admin';
-import {Schema, ResourceCallback} from './common';
+import {Schema, ResourceCallback, LongRunningCallback} from './common';
 
 export type Key = string | string[];
 
-type CommitPromise = Promise<[s.CommitResponse]>;
 export type CreateTableResponse = [
   Table,
   GaxOperation,
   databaseAdmin.longrunning.IOperation
 ];
-type DropTablePromise = Promise<
-  [GaxOperation, databaseAdmin.longrunning.IOperation]
->;
-type ReadPromise = Promise<[Array<Row | Json>]>;
+type DropTableResponse = [GaxOperation, databaseAdmin.longrunning.IOperation];
 
-export interface CreateTableCallback {
-  (
-    err?: ServiceError | null,
-    table?: Table | null,
-    operation?: GaxOperation | null,
-    apiResponse?: databaseAdmin.longrunning.IOperation | null
-  ): void;
-}
+export type CreateTableCallback = LongRunningCallback<Table>;
 
 type DropTableCallback = ResourceCallback<
```

Comment:
Utilize `UpdateSchemaCallback` as well here
```suggestion
export interface DropTableCallback extends UpdateSchemaCallback{};
```

Optional, I would change [`UpdataSchemaCallback`](https://github.com/laljikanjareeya/nodejs-spanner/blob/4d3d64962a1d4c7fafd2b8610209c7aed6b0585b/src/database.ts#L101-L104)
from
```ts
export type UpdateSchemaCallback = ResourceCallback<
  GaxOperation,
  databaseAdmin.longrunning.IOperation
>;
```
to just 
```ts
export interface UpdateSchemaCallback {
  (
    err: ServiceError | null,
    operation?: GaxOperation | null,
    response?: databaseAdmin.longrunning.IOperation
  ): void;
}
```
as `GaxOperation` is not really a `resource`

Reply:
Dave Suggested in `database.ts` PR to use `ResourceCallback` for same.

Reply:
ok